src/hooks/debounce.test.ts

```diff
@@ -0,0 +1,34 @@
+import { renderHook } from '@testing-library/react';
+import { afterEach, beforeAll, describe, expect, it, vi } from 'vitest';
+
+import { useDebouncedFunction } from './debounce';
+
+describe('useDebouncedFunction', () => {
+  beforeAll(() => {
+    vi.useFakeTimers();
+  });
+
+  afterEach(() => {
+    vi.clearAllTimers();
+  });
+
+  it('should debounce function', () => {
+    const myMethod = vi.fn(() => 0);
+    const { result } = renderHook(() => useDebouncedFunction(myMethod, 300));
+    result.current();
+    result.current();
```

Comment:
Why is this called twice?

Reply:
Good question! It does look funny, but this is actually to make sure debounce is doing it's job (calling it twice, but making sure it's executed only once).

Reply:
Ok, makes sense